ui/v2.5/src/components/ScenePlayer/ScenePlayer.tsx

```diff
@@ -479,21 +480,31 @@ export const ScenePlayer: React.FC<IScenePlayerProps> = ({
     const { duration } = file;
     const sourceSelector = player.sourceSelector();
     sourceSelector.setSources(
-      scene.sceneStreams.map((stream) => {
-        const src = new URL(stream.url);
-        const isDirect =
-          src.pathname.endsWith("/stream") ||
-          src.pathname.endsWith("/stream.mpd") ||
-          src.pathname.endsWith("/stream.m3u8");
-
-        return {
-          src: stream.url,
-          type: stream.mime_type ?? undefined,
-          label: stream.label ?? undefined,
-          offset: !isDirect,
-          duration,
-        };
-      })
+      scene.sceneStreams
+        .filter((stream) => {
+          const src = new URL(stream.url);
+          const isFileTranscode =
+            src.pathname.endsWith("/stream.mp4") ||
+            src.pathname.endsWith("/stream.webm");
+          const isSafari = UAParser().browser.name?.includes("Safari");
```

Comment:
There's no need to call this for every stream. This can be moved to line 481.

Reply:
Should be fixed